packages/flutter/test/material/checkbox_test.dart

```diff
@@ -2315,6 +2317,33 @@ void main() {
     );
     expect(getCheckboxRenderer(), paints..path(color: inactiveBackgroundColor));
   });
+
+  testWidgets('Checkbox.adaptive keeps default mouse cursor(Cupertino)', (WidgetTester tester) async {
+    for (final TargetPlatform platform in <TargetPlatform>[ TargetPlatform.iOS, TargetPlatform.macOS ]) {
```

Comment:
I would do this via the `variant` parameter for testWidgets.

Reply:
Is this kind of thing already tested elsewhere on the other platforms?

Reply:
> Is this kind of thing already tested elsewhere on the other platforms?

Yes. There are similar tests in `Switch`, for instance. https://github.com/flutter/flutter/blob/c4ad1dc1e9011d4b42739392e33e4f2cad2fbc2a/packages/flutter/test/material/switch_test.dart#L1235